src/backend/distributed/commands/multi_copy.c

```diff
@@ -821,10 +821,15 @@ OpenCopyConnections(CopyStmt *copyStatement, ShardConnections *shardConnections,
 		ShardPlacement *placement = (ShardPlacement *) lfirst(placementCell);
 		char *nodeUser = CurrentUserName();
 		MultiConnection *connection = NULL;
-		uint32 connectionFlags = FOR_DML | CONNECTION_PER_PLACEMENT;
+		uint32 connectionFlags = FOR_DML;
 		StringInfo copyCommand = NULL;
 		PGresult *result = NULL;
 
+		if (placement->partitionMethod == DISTRIBUTE_BY_HASH)
```

Comment:
This could use a comment.

```
/*
 * Make sure we use a separate connection per placement for hash-distributed
 * tables in order to allow multi-shard modifications in the same transaction.
 */
```